test/query_generator/naming/common.dart

```diff
@@ -0,0 +1,75 @@
+const schema = r'''
+enum Enum {
+  e1
+  e2
+}
+
+input Input {
+  i: String
+  e: Enum
+  s: SubInput
+}
+
+input SubInput {
+  s: String
+}
+
+type Thing {
+  id: String
+  e: Enum
+  aThing: Thing
+  bThing: Thing
+  fThing: Thing
+}
+
+type Query {
+  thing(input: Input!): Thing
+}
+''';
+
+const query = r'''
```

Comment:
I tried to generate the code based on the naming schema - https://tppr.me/RVZp5
I've got 4 identical classes

I tried to replace the entities to fragments and got three classes instead. https://tppr.me/vfrBS

The second case is not ideal but acceptable as far as I can define the variable type inside widget as `ThingIdMixin`

May be we could apply the mixin approach to all classes to reduce amount of possible types?